lib/interceptor.js

```diff
@@ -325,37 +327,37 @@ module.exports = class Interceptor {
       matchKey = common.normalizeOrigin(url)
     }
 
-    if (typeof this.uri === 'function') {
-      matches =
-        common.matchStringOrRegexp(matchKey, this.basePath) &&
-        // This is a false positive, as `uri` is not bound to `this`.
-
-        this.uri.call(this, path)
-    } else {
-      matches =
-        common.matchStringOrRegexp(matchKey, this.basePath) &&
-        common.matchStringOrRegexp(path, this.path)
+    if (!common.matchStringOrRegexp(matchKey, this.basePath)) {
+      const msg = `Base path mismatch: expected ${this.basePath}, got ${matchKey}`
+      this.scope.logger(msg)
+      mismatches.push(msg)
     }
 
-    this.scope.logger(`matching ${matchKey}${path} to ${this._key}: ${matches}`)
+    if (typeof this.uri === 'function') {
+      if (!this.uri.call(this, path)) {
+        const msg = `Path function mismatch: expected function to return true for ${path}`
+        this.scope.logger(msg)
+        mismatches.push(msg)
```

Comment:
If `mismatches` largely mimics the logger, why not use the logger instead to debug request mismatches?

Reply:
Thanks for taking a look! 
Valid point.

Nock's interceptors are stored in an array, and Nock tries each one until it finds a match, so:
1. In logs, it will be easy to miss the relevant interceptor.
2. We can apply logic to the array provided in the listener, for example, sorting it from the best match to the least.